backend/src/payments/payments.controller.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Patch,
+} from '@nestjs/common';
+import { ApiResponse } from '@nestjs/swagger';
+import {
+  PaymentFindAllResponseDto,
+  PaymentFindOneResponseDto,
+  PaymentUpdateResponseDto,
+  UserRole,
+} from '@osk/shared';
+import { Roles } from 'common/guards/roles.decorator';
+import { CurrentUserService } from 'current-user/current-user.service';
+import { assertNever } from 'utils/assertNever';
+import { PaymentsService } from './payments.service';
+
+@Controller('payments')
+export class PaymentsController {
+  constructor(
+    private readonly paymentsService: PaymentsService,
+    private readonly currentUserService: CurrentUserService,
+  ) {}
+
+  @Roles(UserRole.Admin)
+  @ApiResponse({
+    type: PaymentFindAllResponseDto,
+  })
+  @Get()
+  async findAll(): Promise<PaymentFindAllResponseDto> {
+    const payments = await this.paymentsService.findAll();
+
+    return { payments };
+  }
+
+  @Roles(UserRole.Trainee)
+  @ApiResponse({
+    type: PaymentFindAllResponseDto,
+  })
+  @Get('my')
+  async findAllByTrainee(): Promise<PaymentFindAllResponseDto> {
+    const loggedUser = this.currentUserService.getRequestCurrentUser();
+
+    const payments = await this.paymentsService.findAllByTrainee(
+      loggedUser.userId,
+    );
+
+    return { payments };
+  }
+
+  @Roles(UserRole.Trainee)
+  @ApiResponse({
+    type: PaymentFindOneResponseDto,
+  })
+  @Get('my/:id')
+  async findOnePersonalPayment(
+    @Param('id') id: string,
+  ): Promise<PaymentFindOneResponseDto> {
+    const payment = await this.paymentsService.findOnePersonalPayment(+id);
+
+    if (payment.ok) {
+      return { payment: payment.data };
+    }
+
+    if (payment.error === 'PAYMENT_NOT_FOUND') {
+      throw new NotFoundException('Payment with this ID does not exist.');
+    }
+
+    return assertNever(payment.error);
+  }
+
+  @Roles(UserRole.Admin)
+  @ApiResponse({
+    type: PaymentUpdateResponseDto,
+  })
+  @Patch('edit/:id')
+  async editPayment(
+    @Param('id') id: string,
```

Comment:
Let's use `@Param('id', ParseIntPipe) id: number,` instead (then `(+id)` → `(id)` in line 82)